scripts/travis-pr-title-lint

```diff
@@ -0,0 +1,181 @@
+  }
+
+  return commits;
+}
+
+function getGitCommitMessage(commitSha1) {
+  const res = shelljs.exec(`git show -s --format=%B ${commitSha1}`, {silent: true});
+
+  if (res.code !== 0) {
+    throw new Error(`getGitCommitMessage Error: ${res.stderr}`);
+  }
+
+  const commitMessage = res.stdout.trim();
+  if (VERBOSE === 'true') {
+    console.log(`DEBUG getGitCommitMessage: "${commitMessage}"`);
+  }
+
+  return commitMessage;
+}
+
+function getPullRequestTitle() {
+  return new Promise(function(resolve, reject) {
+    const pullRequestURLPath = `/${TRAVIS_REPO_SLUG}/pull/${TRAVIS_PULL_REQUEST}`;
+
+    console.log(`Retrieving the pull request title from https://github.com${pullRequestURLPath}\n`);
+
+    var req = https.get({
+      host: 'github.com',
+      path: pullRequestURLPath,
+      headers: {
+        'User-Agent': 'GitHub... your API can be very annoying ;-)',
+      },
+    }, function(response) {
+      if (response.statusCode < 200 || response.statusCode > 299) {
+        reject(new Error(`getPullRequestTitle got an unexpected statusCode: ${response.statusCode}`));
+        return;
+      }
+
+      var body = '';
+      response.on('data', function(data) {
+        try {
+          body += data;
+
+          // Once we get the closing title tag, we can read the pull request title and
+          // close the http request.
+          if (body.includes('</title>')) {
+            response.removeAllListeners('data');
+            response.emit('end');
+
+            var titleStart = body.indexOf('<title>');
+            var titleEnd = body.indexOf('</title>');
+
+            // NOTE: page slice is going to be something like:
+            // "<title> PR title by author · Pull Request #NUM · mozilla/web-ext · GitHub"
+            var pageTitleParts = body.slice(titleStart, titleEnd)
+              .replace('<title>', '')
+              .split(' · ');
+
+            // Check that we have really got the title of a real pull request.
+            var expectedPart1 = `Pull Request #${TRAVIS_PULL_REQUEST}`;
+
+            if (pageTitleParts[1] === expectedPart1) {
+              // Remove the "by author" part.
+              var prTitleEnd = pageTitleParts[0].lastIndexOf(' by ');
+              resolve(pageTitleParts[0].slice(0, prTitleEnd));
+            } else {
+              console.log('DEBUG getPullRequestTitle response:', body);
+
+              reject(new Error('Unable to retrieve the pull request title'));
+            }
+
+            req.abort();
+          }
+        } catch (err) {
+          reject(err);
+          req.abort();
+        }
+      });
+      response.on('error', function(err) {
+        console.error('Failed during pull request title download: ', err);
+        reject(err);
+      });
+    });
+  });
+}
+
+function lintMessage(message) {
+  if (!message) {
+    throw new Error('Unable to lint an empty message.');
+  }
+
+  const res = shelljs.exec(`echo ${message} | commitlint`, {silent: true});
```

Comment:
There is a command injection vulnerability here.

Consider using commitlint as a library instead of calling it via the CLI: https://github.com/conventional-changelog/commitlint/tree/master/@commitlint/lint

Reply:
ah, yeah, good catch.

I would prefer to don't use commitlint as a library for two reasons:

- I was using the previous changelog-commit-lint as a library, and that part changed a lot and a couple of times between changelog-commit-lint and this new @commitlint/* packages, and because of that we couldn't migrate to @commitlint so easily (because we had to change this script a lot)

- To be able to use @commitlint/* as a library we have to directly depend from something like 3 more @commitlint packages (@commitlint/load, @commitlint/lint and @commitlint/format)

In the updated version of this PR I opted for the following alternative approach:
the arbitrary message is first saved in a temp file, and then the temp file is piped to the commitlint bin script.

This should prevent the message from be able to do any command injection, and we don't have to directly depend from all those @commitlint sub-packages.